ArduPlane/mode_qloiter.cpp

```diff
@@ -3,12 +3,121 @@
 
 bool ModeQLoiter::_enter()
 {
-    return plane.mode_qstabilize._enter();
+    // initialise loiter
+    plane.quadplane.loiter_nav->clear_pilot_desired_acceleration();
+    plane.quadplane.loiter_nav->init_target();
+
+    // set vertical speed and acceleration limits
+    plane.quadplane.pos_control->set_max_speed_accel_z(-plane.quadplane.get_pilot_velocity_z_max_dn(), plane.quadplane.pilot_velocity_z_max_up, plane.quadplane.pilot_accel_z);
+    plane.quadplane.pos_control->set_correction_speed_accel_z(-plane.quadplane.get_pilot_velocity_z_max_dn(), plane.quadplane.pilot_velocity_z_max_up, plane.quadplane.pilot_accel_z);
+
+    plane.quadplane.init_throttle_wait();
+
+    // remember initial pitch
+    plane.quadplane.loiter_initial_pitch_cd = MAX(plane.ahrs.pitch_sensor, 0);
+
+    // prevent re-init of target position
+    plane.quadplane.last_loiter_ms = AP_HAL::millis();
+
+    return true;
 }
 
 void ModeQLoiter::update()
 {
     plane.mode_qstabilize.update();
 }
 
+// run quadplane loiter controller
+void ModeQLoiter::run()
+{
+    if (plane.quadplane.throttle_wait) {
+        plane.quadplane.set_desired_spool_state(AP_Motors::DesiredSpoolState::GROUND_IDLE);
+        plane.quadplane.attitude_control->set_throttle_out(0, true, 0);
+        plane.quadplane.relax_attitude_control();
+        plane.quadplane.pos_control->relax_z_controller(0);
+        plane.quadplane.loiter_nav->clear_pilot_desired_acceleration();
+        plane.quadplane.loiter_nav->init_target();
+        return;
+    }
+    if (!plane.quadplane.motors->armed()) {
+        _enter();
```

Comment:
Ewww.  Factor the guts out into another function if we really need to do this!

The similarities/differences from the `throttle_wait` state are also kind of interesting.